src/utils/index.ts

```diff
@@ -43,14 +67,41 @@ export const downloadTheme = async (
 ) => {
   const theme: DefaultThemeType = { ...defaultTheme, colors: colorMapping };
 
-  const allColors = { ...colorMapping, ...customColorMapping };
+  const allColors: Record<string, string> = {
+    ...colorMapping,
+    ...customColorMapping,
+  };
+  const colors = getPalette(allColors, luminanceSteps);
 
+  const lightSpeakingNames = getSpeakingNamesWithColors(
+    speakingNames,
+    allColors,
+    colors,
+    false,
+    luminanceSteps,
+    speakingNamesDefaultMapping,
+  );
+  const darkSpeakingNames = getSpeakingNamesWithColors(
+    speakingNames,
+    allColors,
+    colors,
+    true,
+    luminanceSteps,
+    speakingNamesDefaultMapping,
+  );
+  console.log("light: ", lightSpeakingNames, "dark: ", darkSpeakingNames);
   const fileName = theme.name || `default-theme`;
   const themeJsonString = JSON.stringify(theme);
+  const colorsJsonString = JSON.stringify({
+    light: lightSpeakingNames,
+    dark: darkSpeakingNames,
+  });
   const themeProperties = getCssThemeProperties(defaultTheme);
 
   const zip = new JSZip();
   zip.file(`${fileName}.json`, themeJsonString);
+  zip.file(`${fileName}-colors.json`, colorsJsonString);
```

Comment:
We should rename it to `sketch-colors` to make it clear which output is generated.